src/main/resources/spring/camel-context.xml

```diff
@@ -51,6 +65,53 @@
             </get>
         </rest>
 
+        <rest path="/administration" produces="application/octet-stream">
+            <get uri="/report/csv">
+                <route id="administration-report-csv">
+                    <setHeader headerName="CSVCorrelationID">
+                        <method ref="utils" method="generateUUID"/>
+                    </setHeader>
+                    <setHeader headerName="unixEpochDate">
+                        <method ref="dateUtils" method="getUnixEpochDate"/>
+                    </setHeader>
+                    <setHeader headerName="nowDate">
+                        <simple>${date:now}</simple>
+                    </setHeader>
+                    <setHeader headerName="fromDate">
+                        <method ref="dateUtils" method="getLastMondayMinus7days"/>
+                    </setHeader>
+                    <setHeader headerName="toDate">
+                        <method ref="dateUtils" method="getLastMonday"/>
+                    </setHeader>
+
+                    <log message="Creating csv first file from date ${date:header.fromDate} to ${date:header.toDate}" />
+                    <bean ref="analysisService" method="getAdministrationMetrics(${header.fromDate}, ${header.toDate})"/>
+                    <bean ref="administrationMetricsMapper" method="toAdministrationMetricsModels"/>
+                    <to uri="direct:administrationMetricsModelToCsv"/>
+                    <setHeader headerName="CamelFileName">
+                        <simple>${date:header.fromDate:yyyyMMddHHmmssz}-to-${date:header.toDate:yyyyMMddHHmmssz}.csv</simple>
+                    </setHeader>
+                    <to uri="direct:administration-report-csv-aggregator" />
+
+                    <log message="Creating csv second file from date ${date:header.fromDate} to ${date:header.toDate}" />
+                    <bean ref="analysisService" method="getAdministrationMetrics(${header.unixEpochDate}, ${header.nowDate})"/>
+                    <bean ref="administrationMetricsMapper" method="toAdministrationMetricsModels"/>
+                    <to uri="direct:administrationMetricsModelToCsv"/>
+                    <setHeader headerName="CamelFileName">
+                        <simple>all.csv</simple>
+                    </setHeader>
+                    <to uri="direct:administration-report-csv-aggregator" />
+
+                    <pollEnrich>
+                        <simple>file:tmpFolder?fileName=${header.CSVCorrelationID}.zip</simple>
```

Comment:
See above in order to avoid to repeat the value `CSVCorrelationID` multiple times

Reply:
@mrizzi Regarding to `CSVCorrelationID`. I don't see any value for creating a constant, in a Java Class; this value will be used just on a couple of routes and it has no meaning for the rest of the routes. If we create a constant for `CSVCorrelationID` why don't we create another constant for the rest of custom headers like `fromDate`, `toDate`?, etc. By the way, I've changed `CSVCorrelationID` to `correlationId` just to be consistent with the lowercase pattern of the rest custom headers.

Reply:
@carlosthe19916 i see your point and i would say that i agree on having also the other customer headers to use constant values.
The benefit seems low when you use them in 2-3 places (even if it means you changed it from `CSVCorrelationID` to `correlationId` 3 times instead of just 1) but it will help the maintenance of the code since changing once is less error prone than changing it n-times.

Reply:
I see your point. My last commit should fix this. Please take a look